src/kudu/consensus/raft_consensus.cc

```diff
@@ -1843,7 +1859,13 @@ Status RaftConsensus::UpdateReplica(const ConsensusRequestPB* request,
       if (op_type == NO_OP) {
         new_leader_detected_failsafe_ = false;
       }
-      prepare_status = StartFollowerTransactionUnlocked(*iter);
+
+      ReplicateMsgWrapper msg_wrapper(*iter, codec_);
+      prepare_status = msg_wrapper.Init(&compression_buffer_);
```

Comment:
Should we allow this bad prepare_status to proceed from here, or is this grounds for RETURN_NOT_OK?

Reply:
I wasn't fully sure how to handle expected_rotation_delay, that's why I went this way. If this fails we'll eventually fail anyway but the semantics of failure will be similar to SFT failing.

Reply:
If decompression fails, the uncompressed msg should be empty. In that case StartFollowerTransaction will not be able to assume its presence. I think we should bail out early if only decompression fails, which is what should be the return code of Init.

Reply:
Not sure if I fully understood. We're not calling SFT if Init() fails (we're checking prepare_status in line 1865 below.

Reply:
yes, my bad!